### PR TITLE
publish pointcloud in ros

### PR DESCRIPTION
**Pull request by sunrentao**
publish pointcloud in ros

### CODE REVIEW

**Comment by ymd-stella** [comment]
Thanks for your contribution!
Please apply clang-format.

File: src/openvslam_ros.cc [hunk 1]
```diff
@@ -60,6 +61,26 @@ void system::publish_pose(const Eigen::Matrix4d& cam_pose_wc, const ros::Time& s
...
+        }
+        const openvslam::Vec3_t pos_w = lm->get_pos_in_world();
+        points.push_back(pcl::PointXYZ(pos_w.z(), -pos_w.x(), -pos_w.y()));
+    }
+    sensor_msgs::PointCloud2 pcout;
+    ROS_INFO("points size is %lu", points.size());
+    pcl::toROSMsg(points, pcout);
+    pcout.header.frame_id = "kdlidar_origin";
+    pc_pub_.publish(pcout);
+
```
Comment (ymd-stella) [inline src/openvslam_ros.cc]: How about changing it as follows?
(The default value for publish_pointcloud_ should be false.)

```suggestion
    if (publish_pointcloud_) {
      std::vector<openvslam::data::landmark*> landmarks;
      std::set<openvslam::data::landmark*> local_landmarks;
      SLAM_.get_map_publisher()->get_landmarks(landmarks, local_landmarks);
      pcl::PointCloud<pcl::PointXYZ> points;
      for (const auto lm : landmarks) {
          if (!lm || lm->will_be_erased()) {
              continue;
          }
          const openvslam::Vec3_t pos_w = rot_ros_to_cv_map_frame * lm->get_pos_in_world();
          points.push_back(pcl::PointXYZ(pos_w.z(), -pos_w.x(), -pos_w.y()));
      }
      sensor_msgs::PointCloud2 pcout;
      pcl::toROSMsg(points, pcout);
      pcout.header.frame_id = map_frame_;
      pc_pub_.publish(pcout);
    }
```

**Comment by sunrentao** [comment]
hi, thanks for suggestion, done

**Comment by ymd-stella** [comment]
LGTM.
Please re-apply clang-format.

**Comment by sunrentao** [comment]
Hi, sorry for last problem, this update should work

File: src/openvslam_ros.cc [hunk 2]
```diff
@@ -60,6 +61,23 @@ void system::publish_pose(const Eigen::Matrix4d& cam_pose_wc, const ros::Time& s
...
+        std::vector<openvslam::data::landmark*> landmarks;
+        std::set<openvslam::data::landmark*> local_landmarks;
+        SLAM_.get_map_publisher()->get_landmarks(landmarks, local_landmarks);
+        pcl::PointCloud<pcl::PointXYZ> points;
+        for (const auto lm : landmarks) {
+            if (!lm || lm->will_be_erased()) {
+                continue;
+            }
+            const openvslam::Vec3_t pos_w = rot_ros_to_cv_map_frame * lm->get_pos_in_world();
+            points.push_back(pcl::PointXYZ(pos_w.z(), -pos_w.x(), -pos_w.y()));
```
Comment (ymd-stella) [inline src/openvslam_ros.cc:74]: I had misread the first code.
If you are converting with `points.push_back(pcl::PointXYZ(pos_w.z(), -pos_w.x(), -pos_w.y()));`, then　`rot_ros_to_cv_map_frame *` is not necessary.
However, `rot_ros_to_cv_map_frame *` is slightly preferable.

**Comment by ymd-stella** [comment]
I noticed that the point cloud is output only while tracking is available because it is implemented in publish_pose.
It would be better to create a new publish_pointcloud method to always output the point cloud.

**Comment by sunrentao** [comment]
ah, sorry, I was not with my working pc in weekend, thank you for making the modification.

**Pull request merged**